### PR TITLE
Deprecate string to atom conversion

### PR DESCRIPTION
**Pull request by msaraiva**
After this gets merge I'll release `v0.4.0`.

### CODE REVIEW

File: lib/surface/components/form/field_context.ex [hunk 1]
```diff
@@ -12,7 +12,7 @@ defmodule Surface.Components.Form.FieldContext do
   alias Surface.Components.Form.Field
 
   @doc "The field name"
-  prop name, :atom, required: true
+  prop name, :any, required: true
```
Comment (msaraiva) [inline lib/surface/components/form/field_context.ex:15]: @miguel-s I believe this change makes it consistent with the changes in https://github.com/surface-ui/surface/pull/319/. Correct? Is there anything else to be done?

File: lib/surface/components/form/label.ex [hunk 2]
```diff
@@ -23,7 +23,7 @@ defmodule Surface.Components.Form.Label do
   prop form, :form
 
   @doc "The field name"
-  prop field, :atom
+  prop field, :any
```
Comment (msaraiva) [inline lib/surface/components/form/label.ex:26]: Same here. Confirm?

**Review by Malian — APPROVED** [review]

File: lib/surface/components/form/field_context.ex (hunk 1 above)
Comment (miguel-s) [reply lib/surface/components/form/field_context.ex:15]: Looks good to me 👍 

We might want to add a test to `field_contest_test.exs` to check `name` with atom, we only have one for strings now.

File: lib/surface/components/form/label.ex (hunk 2 above)
Comment (miguel-s) [reply lib/surface/components/form/label.ex:26]: Also 👍 

Same thing with the test.

**Review by miguel-s — APPROVED** [review]

**Pull request merged**